src/libNode/FinalBlockProcessing.cpp

```diff
@@ -672,8 +672,10 @@ bool Node::ProcessFinalBlock(const bytes& message, unsigned int offset,
     }
   }
 
-  ProcessStateDeltaFromFinalBlock(stateDelta,
-                                  txBlock.GetHeader().GetStateDeltaHash());
+  if (!ProcessStateDeltaFromFinalBlock(
```

Comment:
I think with this check, the node won't RejoinAsNormal anymore, which is called later with CheckStateRoot

```
  if (!LOOKUP_NODE_MODE &&
      (!CheckStateRoot(txBlock) || m_doRejoinAtStateRoot)) {
    RejoinAsNormal();
    return false;
  } else if (LOOKUP_NODE_MODE && !CheckStateRoot(txBlock)) {
    return false;
  }
```